server/jetstream_api.go

```diff
@@ -27,118 +28,121 @@ import (
-	JetStreamTemplateInfo  = "$JS.TEMPLATE.*.INFO"
-	JetStreamTemplateInfoT = "$JS.TEMPLATE.%s.INFO"
+	JSApiTemplateDelete  = "$JS.API.TEMPLATE.DELETE.*"
+	JSApiTemplateDeleteT = "$JS.API.TEMPLATE.DELETE.%s"
 
-	// JetStreamDeleteTemplate is the endpoint to delete stream templates.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamDeleteTemplate  = "$JS.TEMPLATE.*.DELETE"
-	JetStreamDeleteTemplateT = "$JS.TEMPLATE.%s.DELETE"
+	// JSApiCreateStream is the endpoint to create new streams.
+	// Will return JSON response.
+	JSApiStreamCreate  = "$JS.API.STREAM.CREATE.*"
+	JSApiStreamCreateT = "$JS.API.STREAM.CREATE.%s"
 
-	// JetStreamCreateStream is the endpoint to create new streams.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamCreateStream  = "$JS.STREAM.*.CREATE"
-	JetStreamCreateStreamT = "$JS.STREAM.%s.CREATE"
+	// JSApiUpdateStream is the endpoint to update existing streams.
+	// Will return JSON response.
+	JSApiStreamUpdate  = "$JS.API.STREAM.UPDATE.*"
+	JSApiStreamUpdateT = "$JS.API.STREAM.UPDATE.%s"
 
-	// JetStreamUpdateStream is the endpoint to update existing streams.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamUpdateStream  = "$JS.STREAM.*.UPDATE"
-	JetStreamUpdateStreamT = "$JS.STREAM.%s.UPDATE"
+	// JSApiListStreams is the endpoint to list all streams for this account.
+	// Will return JSON response.
+	JSApiStreams = "$JS.API.STREAMS"
 
-	// JetStreamListStreams is the endpoint to list all streams for this account.
-	// Will return json list of string on success and -ERR on failure.
-	JetStreamListStreams = "$JS.STREAM.LIST"
+	// JSApiStreamInfo is for obtaining general information about a named stream.
+	// Will return JSON response.
+	JSApiStreamInfo  = "$JS.API.STREAM.INFO.*"
+	JSApiStreamInfoT = "$JS.API.STREAM.INFO.%s"
 
-	// JetStreamStreamInfo is for obtaining general information about a named stream.
+	// JSApiDeleteStream is the endpoint to delete streams.
 	// Will return JSON response.
-	JetStreamStreamInfo  = "$JS.STREAM.*.INFO"
-	JetStreamStreamInfoT = "$JS.STREAM.%s.INFO"
+	JSApiStreamDelete  = "$JS.API.STREAM.DELETE.*"
+	JSApiStreamDeleteT = "$JS.API.STREAM.DELETE.%s"
 
-	// JetStreamDeleteStream is the endpoint to delete streams.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamDeleteStream  = "$JS.STREAM.*.DELETE"
-	JetStreamDeleteStreamT = "$JS.STREAM.%s.DELETE"
+	// JSApiPurgeStream is the endpoint to purge streams.
+	// Will return JSON response.
+	JSApiStreamPurge  = "$JS.API.STREAM.PURGE.*"
+	JSApiStreamPurgeT = "$JS.API.STREAM.PURGE.%s"
 
-	// JetStreamPurgeStream is the endpoint to purge streams.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamPurgeStream  = "$JS.STREAM.*.PURGE"
-	JetStreamPurgeStreamT = "$JS.STREAM.%s.PURGE"
+	// JSApiDeleteMsg is the endpoint to delete messages from a stream.
+	// Will return JSON response.
+	JSApiMsgDelete  = "$JS.API.MSG.DELETE.*"
+	JSApiMsgDeleteT = "$JS.API.MSG.DELETE.%s"
```

Comment:
this should be `$JS.API.STREAM.something` so it stands out if someone sets out to write ACLs for all things that access a stream same with get

Reply:
I had it that way to start, but realized the only admin for MSG was raw stream. I do still have consumer as a prefix on get next. 

I can add that one back in if you feel strongly.

Reply:
Fixed delete and get.